src/build_context.rs

```diff
@@ -73,22 +73,31 @@ pub enum ProjectLayout {
 
 impl ProjectLayout {
     /// Checks whether a python module exists besides Cargo.toml with the right name
-    pub fn determine(project_root: impl AsRef<Path>, module_name: &str) -> Result<ProjectLayout> {
+    pub fn determine(
+        project_root: impl AsRef<Path>,
+        module_name: &str,
+        py_src: Option<impl AsRef<Path>>,
+    ) -> Result<ProjectLayout> {
         // A dot in the module name means the extension module goes into the module folder specified by the path
         let parts: Vec<&str> = module_name.split('.').collect();
         let project_root = project_root.as_ref();
+        let python_root = if let Some(py_src) = py_src {
+            project_root.join(py_src.as_ref())
+        } else {
+            project_root.to_path_buf()
+        };
```

Comment:
To avoid the possible copy:
```suggestion
        let python_root = py_src.map_or(
            Cow::Borrowed(project_root),
            |py_src| Cow::Owned(project_root.join(py_src))
        );
```

Reply:
These copies are mostly one time cost, I don't think they worth the complexity.